app/components/Body.tsx

```
@@ -0,0 +1,44 @@
+import * as React from 'react'
+import { ApiAction } from '../store/api'
+import HandsonTable from './HandsonTable'
+import { WorkingDataset } from '../models/store'
+
+interface BodyProps {
+  workingDataset: WorkingDataset
+  fetchBody: () => Promise<ApiAction>
+}
+
+export default class Body extends React.Component<BodyProps> {
+  constructor (props: BodyProps) {
+    super(props)
+
+    this.state = {}
+  }
+
+  static getDerivedStateFromProps (nextProps: BodyProps) {
+    const { workingDataset } = nextProps
+    const { value, bodyLoading: isLoading } = workingDataset
+    const { body } = value
+    if (
+      body === undefined &&
+      isLoading === false
```

If body is empty, will it come back as [] or {}?

Also, if body is undefined and isLoading is false because of an error, and the error is never handled or reacted to, this will become an infinite loop.

---

I think we should punt on this for now, and take a closer look at error states and notifications for all of our api calls together soon.  Opened an issue for this: #27